pmoves/services/hi-rag-gateway-v2/routes/geometry.py

```
@@ -0,0 +1,574 @@
+                raise HTTPException(502, f"failed to fetch image: {e}")
+            image_bytes = r.data
+            if not image_bytes:
+                raise HTTPException(400, f"remote image returned empty body for {url}")
+            try:
+                img = Image.open(io.BytesIO(image_bytes)).convert('RGB')
+            except Exception as e:
+                raise HTTPException(400, f"invalid image payload for {url}: {e}")
+            img_list.append(img)
+        img_embs = model.encode(img_list, normalize_embeddings=True, convert_to_numpy=True)
+        sims = (img_embs @ text_emb.T).squeeze()
+        ranked = sorted(zip(images, sims.tolist()), key=lambda x: x[1], reverse=True)
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)